routing/router.go

```diff
@@ -1907,3 +1937,65 @@ func (r *ChannelRouter) AddProof(chanID lnwire.ShortChannelID,
 	info.AuthProof = proof
 	return r.cfg.Graph.UpdateChannelEdge(info)
 }
+
+// IsStaleNode returns true if the graph source has a node announcement for the
+// target node with a more recent timestamp.
+//
+// NOTE: This method is part of the ChannelGraphSource interface.
+func (r *ChannelRouter) IsStaleNode(node Vertex, timestamp time.Time) bool {
+	// If our attempt to assert that the node announcement is fresh fails,
+	// then we know that this is actually a stale announcement.
+	return r.assertNodeAnnFreshness(node, timestamp) != nil
```

Comment:
If the node does not exist, then this will fail?

Reply:
Yep, as we need the _partial_ node announcement to be in the channel graph (added after we get a chan ann) before we'll accept the full node announcement. We'll insert a partial node when we get the channel announcement, and the full node (addrs, etc) once we get the chan ann.

Reply:
This is counterintuitive imo, as the godoc states that the method returns true if we have already seen a newer node announcement, but the real behaviour is that it returns true if `(we have seen newer node announcement || we have never seen a channel announcement from this node)`.

Either the godoc and method names should be updated, or the behaviour should be changed to reflect the doc and usage be 

```go
onReceiveNodeAnnouncement(ann) {
    if !existsInGraph(ann) {
        return ignoreNode
    }

    if IsStaleNode(ann) {
        return ignoreNode
    }

    apply(ann)
}
```

Reply:
Fail as in return true? Yes